ConsoleCreateSpanAnnotation/Program.cs

```diff
@@ -0,0 +1,134 @@
+            var contained = new IfcRelContainedInSpatialStructure(site);
+            contained.RelatedElements.Add(annotation);
+
+            #region Alignment
+
+            var alignment = new IfcAlignment(site)
+            {
+                Name = "Basic alignment with a single horizontal segment"
+            };
+            alignment.AddComment("Generate alignment representation");
+
+            // semantic
+            var horizSegment = new IfcAlignmentHorizontalSegment(
+                new IfcCartesianPoint(database, 5, 10),
+                0,
+                0,
+                0,
+                200,
+                IfcAlignmentHorizontalSegmentTypeEnum.LINE);
+
+            // geometric representation of a single segment. It gets referenced by IfcCompositeCurve.segments and IfcAlignmentSegment.Representation
+            var curveSegment = new IfcCurveSegment(
+                 IfcTransitionCode.CONTSAMEGRADIENTSAMECURVATURE,
+                 new IfcAxis2Placement2D(new IfcCartesianPoint(database, 0, 0)),
+                 new IfcParameterValue(0),
+                 new IfcParameterValue(200),
+                 null);
+
+
+            var segments = new List<IfcSegment> { curveSegment };
+            var compositeCurve = new IfcCompositeCurve(segments);
+
+            var rep = new IfcShapeRepresentation(compositeCurve)
+            {
+                RepresentationIdentifier = "Axis",
+                RepresentationType = "Curve2D"
+            };
+
+            alignment.Representation = new IfcProductDefinitionShape(rep);
+            alignment.Axis = compositeCurve;
+
+            // create an alignment horizontal instance that takes a list of horizontal segments
+            var alignmentHorizontal = new IfcAlignmentHorizontal(database)
+            {
+                Segments = new LIST<IfcAlignmentHorizontalSegment>() { horizSegment }
+            };
+
+            // link alignment and and its horizontal part semantically
+            new IfcRelAggregates(alignment, alignmentHorizontal);   
+
+            // create a new alignmentSegment with then gets one curve segment as its geometric representation
+            var alignmentSegment = new IfcAlignmentSegment(database);
+
+            // link horizontal alignment with the recently created alignment segment
+            new IfcRelNests(alignmentHorizontal, alignmentSegment); // sorted list -> IfcRelNests
+            
+            // connect geom representation to this segment
+            alignmentSegment.Representation = new IfcProductDefinitionShape(new IfcShapeRepresentation(curveSegment));
+
+            #endregion Alignment
+
+            #region Annotation placement
+
+            alignmentSegment.AddComment("Create placement for annotation");
+
+            var axis2place = new IfcAxis2PlacementLinear(
+                new IfcPointByDistanceExpression(25, compositeCurve),
+                null,
+                null);
```

Comment:
Axis2PlacementLinear needs a reference to a curve along which it is defined. (`compositeCurve` probably in your case)